guvnor-asset-mgmt/guvnor-asset-mgmt-client/src/test/java/org/guvnor/asset/management/client/editors/repository/wizard/pages/RepositoryInfoPageTest.java

```diff
@@ -0,0 +1,280 @@
+
+        assertEquals( organizationalUnits.get( 0 ), model.getOrganizationalUnit() );
+
+        assertPageComplete( false, infoPage );
+    }
+
+    /**
+     * Tests that the page reacts properly when a valid repository name is entered.
+     */
+    @Test
+    public void testValidRepositoryNameChange( ) {
+
+        RepositoryInfoPage infoPage = new RepositoryInfoPageExtended( view,
+                new OrganizationalUnitServiceCallerMock( organizationalUnitService ),
+                new RepositoryServiceCallerMock( repositoryService ),
+                true,
+                new WizardTestUtils.WizardPageStatusChangeEventMock() );
+
+        CreateRepositoryWizardModel model = new CreateRepositoryWizardModel();
+        infoPage.setModel( model );
+
+        when( repositoryService.validateRepositoryName( "ValidRepo" ) ).thenReturn( true );
+        when( repositoryService.validateRepositoryName( "InvalidRepo" ) ).thenReturn( false );
+
+
+        when( view.getName() ).thenReturn( "ValidRepo" );
+        infoPage.onNameChange();
+
+        verify( view, times( 2 ) ).getName();
+        verify( view, times( 1 ) ).clearNameErrorMessage();
+
+
+        assertEquals( "ValidRepo", model.getRepositoryName() );
+
+        assertPageComplete( false, infoPage );
+    }
+
+    /**
+     * Tests that the page reacts properly when an invalid repository name is typed.
+     */
+    @Test
+    public void testInvalidRepositoryNameChange( ) {
+
+        RepositoryInfoPage infoPage = new RepositoryInfoPageExtended( view,
+                new OrganizationalUnitServiceCallerMock( organizationalUnitService ),
+                new RepositoryServiceCallerMock( repositoryService ),
+                true,
+                new WizardTestUtils.WizardPageStatusChangeEventMock() );
+
+        CreateRepositoryWizardModel model = new CreateRepositoryWizardModel();
+        infoPage.setModel( model );
+
+        when( repositoryService.validateRepositoryName( "ValidRepo" ) ).thenReturn( true );
+        when( repositoryService.validateRepositoryName( "InvalidRepo" ) ).thenReturn( false );
+
+
+        when( view.getName() ).thenReturn( "InvalidRepo" );
+        infoPage.onNameChange();
+
+        verify( view, times( 2 ) ).getName();
+        verify( view, times( 1 ) ).setNameErrorMessage( anyString() );
+
+        assertEquals( "InvalidRepo", model.getRepositoryName() );
+
+        assertPageComplete( false, infoPage );
+    }
+
+    /**
+     * Tests that the page reacts properly when the managed repository option is checked.
+     */
+    @Test
+    public void testManagedRepositorySelected() {
+        testManagedRepositoryChange( true );
+    }
+
+    /**
+     * Tests that the page reacts properly when the managed repository option is un checked.
+     */
+    @Test
+    public void testUnManagedRepositorySelected() {
+        testManagedRepositoryChange( false );
+    }
+
+    public void testManagedRepositoryChange( boolean isManaged ) {
```

Comment:
And just a nitpick here: I'd make this method private to show that it's just a 'test helper', not a test method itself.
Otherwise, Kudos for nice coverage!